File: ccmtools/test/CppGenerator/assembly_nested/impl/assembly/assembly_factory.cc
#include "assembly.h"
#include "assembly_factory.h"

#include <wx/utils/debug.h>

using namespace std;
using namespace wx::utils;

namespace ccm {
namespace local {

AssemblyFactory::AssemblyFactory()
{
  DEBUGNL("+AssemblyFactory::AssemblyFactory()");
}
AssemblyFactory::~AssemblyFactory()
{
  DEBUGNL("-AssemblyFactory::~AssemblyFactory()");
}

wx::utils::SmartPtr<Components::ccm::local::Assembly>
AssemblyFactory::create()
  throw(Components::ccm::local::CreateFailure)
{
  DEBUGNL(" AssemblyFactory::create()");
  wx::utils::SmartPtr<Components::ccm::local::Assembly>
    assembly(new Assembly());
  return assembly;
}

} // /namespace ccm
} // /namespace local

File: ccmtools/test/JavaLocalGenerator/component_assembly_nested/impl/AssemblyFactory.java
package world;

import ccm.local.Components.CreateFailure;


public class AssemblyFactory
    implements ccm.local.Components.AssemblyFactory 
{
    public ccm.local.Components.Assembly create()
	throws CreateFailure
    {
	System.out.println(" AssemblyFactory.create()");
	return new world.Assembly();
    }
}
